docs/reference/typing-faq.md

```diff
@@ -0,0 +1,176 @@
+
+circle_area(2)      # OK: int is allowed where float is expected
+```
+
+This rule is a special case, since `int` is not actually a subclass of `float`. To support this, ty
+treats `float` annotations as meaning `int | float`. Unlike some other type checkers, ty makes this
+behavior explicit in type hints and error messages. For example, if you
+[hover over the `radius` parameter](https://play.ty.dev/fdc144c6-031c-4af9-b520-a4c6ccde9261), ty
+will show `int | float`.
+
+A similar rule applies to `complex`, which is treated as `int | float | complex`.
+
+!!! info
+
+    These special rules for `float` and `complex` exist for a reason. In almost all cases, you
+    probably want to accept both `int` and `float` when you annotate something as `float`.
+    If you really need to accept *only* `float` and not `int`, you can use ty's `JustFloat`
+    type. At the time of writing, this import needs to be guarded by a `TYPE_CHECKING` block:
+
+    ```py
+    from typing import TYPE_CHECKING
+
+    if TYPE_CHECKING:
+        from ty_extensions import JustFloat
+    else:
+        JustFloat = float
+
+    def only_actual_floats_allowed(f: JustFloat) -> None: ...
+
+    only_actual_floats_allowed(1.0)  # OK
+    only_actual_floats_allowed(1)    # error: invalid-argument-type
+    ```
+
+    ([Full example in the playground](https://play.ty.dev/fb034780-3ba7-4c6a-9449-5b0f44128bab))
+
+    If you need this for `complex`, you can use `ty_extensions.JustComplex` in a similar way.
+
+## Does ty have a strict mode?
+
+Not yet. A stricter inference mode is tracked in
+[this issue](https://github.com/astral-sh/ty/issues/1240). In the meantime, you can consider using Ruff's
+[`flake8-annotations` rules](https://docs.astral.sh/ruff/rules/#flake8-annotations-ann) to enforce
+more explicit type annotations in your code.
+
+## Why can't ty resolve my imports?
+
+Import resolution issues are often caused by a missing or incorrect environment configuration. When
+ty reports *"Cannot resolve imported module …"*, check the following:
+
+1. **Virtual environment**: Make sure your virtual environment is discoverable. ty looks for an
+    active virtual environment via `VIRTUAL_ENV` or a `.venv` directory in your project root. See the
+    [module discovery](../modules.md#python-environment) documentation for more details.
+
+1. **Project structure**: If your source code is not in the project root or `src/` directory,
+    configure [`environment.root`](./configuration.md#root) in your `pyproject.toml`:
+
+    ```toml
+    [tool.ty.environment]
+    root = ["./app"]
+    ```
+
+1. **Third-party packages**: Ensure dependencies are installed in your virtual environment. Run ty
+    with `-v` to see the search paths being used.
+
+1. **Compiled extensions**: ty requires `.py` or `.pyi` files for type information. If a package
+    contains only compiled extensions (`.so` or `.pyd` files), you'll need stub files (`.pyi`) for ty
+    to understand the types.
```

Comment:
This is not my area of expertise. Can someone proof-read and possibly extend this?